### ISSUE TITLE
Error causing pages to not display

### ISSUE THREAD

**Issue by allanaaa** [omeka-s-modules/Hierarchy#6]
There seems to be an error with the page block when logged-out - I can see the whole page fine from my admin account, but on a private browser I'm seeing:

 

> Omeka\Api\Exception\NotFoundException
> Omeka\Entity\ItemSet entity with criteria {"id":4146} not found
> 
> Details:
> 
> Omeka\Api\Exception\NotFoundException: Omeka\Entity\ItemSet entity with criteria {"id":4146} not found in /var/www/html/amayer/amayer-s/omeka-s/application/src/Api/Adapter/AbstractEntityAdapter.php:722
> Stack trace:
> #0 /var/www/html/amayer/amayer-s/omeka-s/application/src/Api/Adapter/AbstractEntityAdapter.php(440): Omeka\Api\Adapter\AbstractEntityAdapter->findEntity(Array, Object(Omeka\Api\Request))
> #1 /var/www/html/amayer/amayer-s/omeka-s/application/src/Api/Manager.php(230): Omeka\Api\Adapter\AbstractEntityAdapter->read(Object(Omeka\Api\Request))
> #2 /var/www/html/amayer/amayer-s/omeka-s/application/src/Api/Manager.php(115): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
> #3 /var/www/html/amayer/amayer-s/omeka-s/application/src/View/Helper/Api.php(69): Omeka\Api\Manager->read('item_sets', 4146, Array)
> #4 /var/www/html/amayer/amayer-s/omeka-s/modules/Hierarchy/view/hierarchy/common/block-layout/hierarchy-public.phtml(31): Omeka\View\Helper\Api->read('item_sets', 4146)
> #5 /var/www/html/amayer/amayer-s/omeka-s/modules/Hierarchy/view/hierarchy/common/block-layout/hierarchy-public.phtml(49): Laminas\View\Renderer\PhpRenderer->{closure}(Array)
> #6 /var/www/html/amayer/amayer-s/omeka-s/modules/Hierarchy/view/hierarchy/common/block-layout/hierarchy-public.phtml(54): Laminas\View\Renderer\PhpRenderer->{closure}(Array)
> #7 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php(519): include('/var/www/html/a...')
> #8 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/Helper/Partial.php(63): Laminas\View\Renderer\PhpRenderer->render(NULL, NULL)
> #9 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php(407): Laminas\View\Helper\Partial->__invoke('hierarchy/commo...', Array)
> #10 /var/www/html/amayer/amayer-s/omeka-s/modules/Hierarchy/src/Site/BlockLayout/Hierarchy.php(73): Laminas\View\Renderer\PhpRenderer->__call('partial', Array)
> #11 /var/www/html/amayer/amayer-s/omeka-s/application/src/View/Helper/BlockLayout.php(158): Hierarchy\Site\BlockLayout\Hierarchy->render(Object(Laminas\View\Renderer\PhpRenderer), Object(Omeka\Api\Representation\SitePageBlockRepresentation))
> #12 /var/www/html/amayer/amayer-s/omeka-s/application/src/View/Helper/PageLayout.php(85): Omeka\View\Helper\BlockLayout->render(Object(Omeka\Api\Representation\SitePageBlockRepresentation))
> #13 /var/www/html/amayer/amayer-s/omeka-s/application/view/omeka/site/page/content.phtml(1): Omeka\View\Helper\PageLayout->render(Object(Omeka\Api\Representation\SitePageRepresentation))
> #14 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php(519): include('/var/www/html/a...')
> #15 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/View.php(194): Laminas\View\Renderer\PhpRenderer->render(NULL)
> #16 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/View.php(222): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
> #17 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/View.php(187): Laminas\View\View->renderChildren(Object(Laminas\View\Model\ViewModel))
> #18 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/View.php(222): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
> #19 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-view/src/View.php(187): Laminas\View\View->renderChildren(Object(Laminas\View\Model\ViewModel))
> #20 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-mvc/src/View/Http/DefaultRenderingStrategy.php(98): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
> #21 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-eventmanager/src/EventManager.php(319): Laminas\Mvc\View\Http\DefaultRenderingStrategy->render(Object(Laminas\Mvc\MvcEvent))
> #22 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-eventmanager/src/EventManager.php(171): Laminas\EventManager\EventManager->triggerListeners(Object(Laminas\Mvc\MvcEvent))
> #23 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-mvc/src/Application.php(360): Laminas\EventManager\EventManager->triggerEvent(Object(Laminas\Mvc\MvcEvent))
> #24 /var/www/html/amayer/amayer-s/omeka-s/vendor/laminas/laminas-mvc/src/Application.php(341): Laminas\Mvc\Application->completeRequest(Object(Laminas\Mvc\MvcEvent))
> #25 /var/www/html/amayer/amayer-s/omeka-s/index.php(21): Laminas\Mvc\Application->run()
> #26 {main}

I guess because the item set isn't added to the site as a resource - but this should probably just skip it instead of taking down the whole page.

**Comment by allanaaa** [comment]
Oh, it's because one of the item sets was set to "private" - should've guessed that from the logged-in/out bit.

**Comment by matthewjmckinley** [comment]
Hmmm that's very odd, I have that API call in a try/catch block that is catching that NotFoundException specifically. It's working fine on my machine, even when logged out. 

What does your hierarchy structure/site hierarchy settings look like? Are there multiple private item sets or just the one?

**Comment by zerocrates** [comment]
@matthewjmckinley are you sure you've tested a path that would cause the exception? You'd have to both have the set private and try to view it when not logged in.

That catch isn't fully qualified and there's no `use` so I don't think it catches anything really.

**Comment by matthewjmckinley** [comment]
I was logged out and had the item set as private, but I guess I was still able to see the page because the grouping with the private item set assigned did not have a label, and in those cases the itemSet value gets cleared out earlier on in the HierarchyHelper so it never reached the try/catch block in the view. This must have been the case in my earlier testing which is why I missed the useless catch.

In any case, I've added a use statement so this bug should be fixed, and this also led me to a few other display bugs that have been fixed in the latest commits. Thanks for bringing this to my attention!

If everything looks good I will push a new bugfix release next week. If not we can re-open this issue.
